<?xml version="1.0"?>
<launch>
    <!-- vehicle model and world -->
    <arg name="est" default="ekf2"/>
    <arg name="vehicle" default="iris"/>
    <arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/empty.world"/>
    
    <!-- gazebo configs -->
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>
    
    <!-- Gazebo sim -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world_name" value="$(arg world)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
    </include>
    
    
    
    <!-- UAV0 -->
    <group ns="uav0">
    
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="0"/>
        <arg name="fcu_url" default="udp://:14540@localhost:14580"/>
        
        <!-- Changing the GPS position (Will affect QGC location) -->
        <arg name="mavros/global_position/global/latitude" value="-27.415"/>
        <arg name="mavros/global_position/global/longitude" value="153.1"/>
        <arg name="mavros/global_position/global/altitude" value="0"/>
        
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn.launch">
        
            <arg name="x" value="-2"/>
            <arg name="y" value="1"/>
            <arg name="z" value="0"/>
            
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14560"/>
            <arg name="mavlink_tcp_port" value="4560"/>
            <arg name="ID" value="$(arg ID)"/>
            <arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))"/>
            <arg name="video_uri" value="$(eval 5600 + arg('ID'))"/>
            <arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))"/>
        </include>
        
        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>
        
        <node pkg="project" type="drone1_controller" name="drone1_controller_node" output="screen">
            <param name="fcu_url" value="$(arg fcu_url)"/>
            <param name="gcs_url" value=""/>
            
            <param name="init_x" value="-2"/>
            <param name="init_y" value="1"/>
            <param name="init_z" value="0"/>
            
            <param name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <param name="tgt_component" value="1"/>
        </node>
    </group>
    
    
    
    <!-- UAV1 -->
    <group ns="uav1">
    
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="1"/>
        <arg name="fcu_url" default="udp://:14541@localhost:14581"/>
        
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn.launch">
        
            <arg name="x" value="-2"/>
            <arg name="y" value="-3"/>
            <arg name="z" value="0"/>
            
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14561"/>
            <arg name="mavlink_tcp_port" value="4561"/>
            <arg name="ID" value="$(arg ID)"/>
            <arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))"/>
            <arg name="video_uri" value="$(eval 5600 + arg('ID'))"/>
            <arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))"/>
        </include>
        
        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>
        
        <node pkg="project" type="drone2_controller" name="drone2_controller_node" output="screen">
            <param name="fcu_url" value="$(arg fcu_url)"/>
            <param name="gcs_url" value=""/>
            
            <param name="init_x" value="-2"/>
            <param name="init_y" value="-3"/>
            <param name="init_z" value="0"/>
            
            <param name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <param name="tgt_component" value="1"/>
        </node>
    </group>
    
    
    
    <!-- UAV2 -->
    <group ns="uav2">
    
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="2"/>
        <arg name="fcu_url" default="udp://:14542@localhost:14582"/>
        
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn.launch">
        
            <arg name="x" value="1"/>
            <arg name="y" value="2"/>
            <arg name="z" value="0"/>
            
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14562"/>
            <arg name="mavlink_tcp_port" value="4562"/>
            <arg name="ID" value="$(arg ID)"/>
            <arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))"/>
            <arg name="video_uri" value="$(eval 5600 + arg('ID'))"/>
            <arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))"/>
        </include>
        
        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>
        
        <node pkg="project" type="drone3_controller" name="drone3_controller_node" output="screen">
            <param name="fcu_url" value="$(arg fcu_url)"/>
            <param name="gcs_url" value=""/>
            
            <param name="init_x" value="1"/>
            <param name="init_y" value="2"/>
            <param name="init_z" value="0"/>
            
            <param name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <param name="tgt_component" value="1"/>
        </node>
    </group>
    
    
    
    <node pkg="project" type="collision_avoidance" name="collision_avoidance_node" output="screen"/>
</launch>

